### ISSUE TITLE
buttons rake not updating dialog_label 

### ISSUE THREAD

**Issue by marc-torres** [rhtconsulting/cfme-rhconsulting-scripts#52]
When I attempt to update an existing button with the button rake, I can clearly see the dialog_label is correctly set in the buttons.yml I'm about to rake and import into cloudforms, and the debug output properly shows the correct dialog.id , but when the import is finished, only the 'Hover text' is updated. The dialog_label is never updated with what I have in my buttons.yml.

**Comment by branic** [comment]
@marc-torres can you provide an example and steps to reproduce? Also, what version of CloudForms or ManageIQ is being used?

**Comment by marc-torres** [comment]
@branic details below! 

Cloudforms 3.2 on RHEL6

Issue Description:
Rake script is importing changes into Cloudforms. For 1 particular button (Delete VM button) , when I run the rake, it fails to update the 'dialog label' for this particular existing button. 

The Delete VM button was a pre-existing button although I had to change the dialog that it referenced. THe old dialog pointed to 'Remedy Ticket'. My new changes (that I'm raking in ) has the dialog set to 'Delete VM'.

I can see when I run the rake, in debug mode that it's properly setting the right element in the array that's getting sent. The 'HOver text' is updating for the button, but the dialog label is not. Which leads me to believe its only that element in particular that fails to update.

Unfortunately I cannot post screen shots but they are available through a Red Hat case. Would you like for me to post the case # ?

**Comment by branic** [comment]
@marc-torres directly send me the case # and I'll take a look (bevans@redhat.com)

**Comment by branic** [comment]
@marc-torres thanks for the email. I've had a look at the case.

The dialog_label (Dialog field in the UI) is the name of the Service Dialog to display when the button is clicked.

Do you have a Service Dialog named 'Delete VM'? In the Web UI do you see the service dialog in the Dialog drop-down when editing the button?

I haven't been able to reproduce this on my 3.2 instance. If the Service Dialog exists the Dialog field is updated. If it doesn't exist I get a "Unable to locate dialog" error.

**Comment by marc-torres** [comment]
Yes , I have a 'delete vm' button. It used to be named 'remedy ticket'. So I changed it to reflect 'delete vm' then exported the configuration , then imported it into another environment which already had the 'remedy ticket' button. When I do so, the dialog_label does not get updated to reflect "delete vm' , it stays "remedy ticket". Hope that clears it up



		_____________________________

**Comment by branic** [comment]
The Dialog field (dialog_label in the YAML file) controls which Service Dialog to display (if any) when the button is clicked (on a VM in this case). Service Dialogs are different than buttons and are found on Service Dialogs accordian (just above the Buttons accordian).

Can you email me a screenshot of what the button looks like in the environment that the export comes from?

When performing the imports are the Service Dialogs being imported before the Buttons?

**Comment by wce3** [comment]
Is this issue still open? We are hitting it with button imports with cfme 5.6.3.3. 
We're using the scripts from commit 81b8bad44 (Jul 21 2016)
I can provide Details how it occurs on our Systems.
